Content.Server/Disease/Effects/DiseaseAddComp.cs

```diff
@@ -0,0 +1,22 @@
+using Content.Shared.Disease;
+using JetBrains.Annotations;
+
+namespace Content.Server.Disease.Effects;
+
+[UsedImplicitly]
+public sealed class DiseaseAddComp : DiseaseEffect
+{
+    [DataField("comp")]
+    public string? Comp = null;
+    public override void Effect(DiseaseEffectArgs args)
+    {
+        if (Comp == null) return;
+
+        EntityUid uid = args.DiseasedEntity;
+        Component newComponent = (Component) IoCManager.Resolve<IComponentFactory>().GetComponent(Comp);
+        newComponent.Owner = uid;
+
+        if (!args.EntityManager.HasComponent(uid, newComponent.GetType()))
```

Comment:
Can't this check just be replaced with an EnsureComponent

Reply:
I don't think that work, because you won't know what comp you will need to ensure. So you can't pass dwn  tyoe

Reply:
You can add a new method for it.

Reply:
Okay zombie rework gonna deprecate this anyway so nvm.